yt_dlp/extractor/icareus.py

```diff
@@ -0,0 +1,205 @@
+        'only_matching': True
+    }, {
+        'url': 'https://www.permanto.fi/fi/web/alfatv/player/vod?assetId=135497515',
+        'only_matching': True
+    }, {
+        'url': 'https://videos.minifiddlers.org/web/international-minifiddlers/player/vod?assetId=1982759',
+        'only_matching': True
+    }]
+
+    def _real_extract(self, url):
+        base_url, maybe_id = self._match_valid_url(url).groups()
+        page = self._download_webpage(url, maybe_id)
+        video_id = self._search_regex(
+            r"_icareus\['itemId'\]\s*=\s*'(\d+)'", page, "video_id")
+        api_base = self._search_regex(
+            r'var\s+publishingServiceURL\s*=\s*"(http[^"]+)";', page, "api_base")
+        organization_id = self._search_regex(
+            r"_icareus\['organizationId'\]\s*=\s*'(\d+)'", page, "organization_id")
+        token = self._search_regex(
+            r"_icareus\['token'\]\s*=\s*'([a-f0-9]+)'", page, "token")
+        token2 = self._search_regex(
+            r"""data\s*:\s*{action:"getAsset".*?token:'([a-f0-9]+)'}""", page,
+            "token2", default=None, fatal=False)
+        livestream_title = get_element_by_class(
+            'unpublished-info-item future-event-title', page)
+        metad = self._search_json_ld(page, video_id, default=None)
+
+        duration = None
+        thumbnail = None
+        if metad:
+            title = metad.get('title')
+            description = metad.get('description')
+            timestamp = metad.get('timestamp')
+            thumbnail = traverse_obj(metad, ('thumbnails', 0, 'url'))
+        elif token2:
+            data = {
+                "version": "03",
+                "action": "getAsset",
+                "organizationId": organization_id,
+                "assetId": video_id,
+                "languageId": "en_US",
+                "userId": "0",
+                "token": token2,
+            }
+            metad = self._download_json(
+                base_url + '/icareus-suite-api-portlet/publishing',
+                video_id, data=urlencode_postdata(data))
+            title = metad.get('name')
+            description = metad.get('description')
+            timestamp = int_or_none(metad.get('date'), scale=1000)
+            duration = int_or_none(metad.get('duration'))
+            thumbnail = url_or_none(metad.get('thumbnailMedium'))
+        elif livestream_title:  # Recorded livestream
+            title = livestream_title
+            description = get_element_by_class(
+                'unpublished-info-item future-event-description', page)
+            timestamp = int_or_none(self._search_regex(
+                r"var startEvent\s*=\s*(\d+);", page, "uploadDate",
+                fatal=False), scale=1000)
+        else:
+            self.report_warning("Could not extract metadata", video_id)
+            description = None
+            timestamp = None
+
+        title = title if title else video_id
```

Comment:
why?

Reply:
There's a chance that metadata extraction fails, but the formats can still be parsed. Should the extractor fail without a real title?

Reply:
setting title to None is non-fatal

Reply:
Ok, thanks, this should be ok now.